tools/util/include/cutlass/util/reference/device/sycl_tensor_fill.h

```diff
@@ -114,20 +114,20 @@ struct RandomUniformFunc {
   /// Parameters object
   Params params;
   oneapi::mkl::rng::device::uniform<FloatType> distribution;
+  oneapi::mkl::rng::device::philox4x32x10<> generator;
 
   //
   // Methods
   //
 
   explicit RandomUniformFunc(Params const &params):
       params(params),
-      distribution(static_cast<FloatType>(params.min), static_cast<FloatType>(params.max)){}
+      distribution(static_cast<FloatType>(params.min), static_cast<FloatType>(params.max)),
+      generator(params.seed, ThreadIdxX() + BlockIdxX() * BlockDimX()) {}
```

Comment:
should the offset also be multiplied by the number of values per thread?
Otherwise I imagine that the second value from one thread is the first from the next one.

Reply:
:fire: great catch! Unfortunately `RandomUniformFunc` doesn't have visibility of how many times it'll be called. I've changed the offset to `{0, ThreadIdxX() + BlockIdxX() * BlockDimX()}`. According to [the docs](https://www.intel.com/content/www/us/en/docs/onemkl/developer-reference-dpcpp/2023-0/oneapi-mkl-rng-device-philox4x32x10.html), this means each thread will be offset 2^64 values into the sequence compared with its neighbour. Philox has a [period](https://numpy.org/doc/2.1/reference/random/bit_generators/philox.html) of `2^256 - 1`, so this should be safe!